File: web/js/controllers/UserController.js
'use strict';

controllers.controller('UserController', function ($scope, $http, UserService) {
    $scope.users = {};
    $scope.user = {};

    $scope.getUsers = function(){
        UserService.getUsers().then(function (users) {
            console.log(users);
            $scope.users = users;
        }, function (err) {
            console.log("getUsers ERROR ", err);
        });
    };

    $scope.getUser = function () {
        UserService.getUser().then(function (user) {
            console.log(user);
            $scope.user = user;
        }, function (err) {
            console.log("getUser ERROR ", err);
        });
    };

    $scope.updateUser = function (user) {
        UserService.updateUser(user).then(function (user) {
            console.log(user);
            $scope.user = user;
        }, function (err) {
            console.log("updateUser ERROR ", err);
        });
    };

});

//controllers.controller('UserController', ['$scope', '$dialogs', 'UsersService',
//    function ($scope, $dialogs, UsersService) {
//
//        $scope.data = {};
//
//        getUsers();
//
//        var getUsers = function () {
//
//            UsersService.getUsers(function (successResult) {
//                $scope.users = successResult;
//            },
//            function (errorResult) {
//                $dialogs.error("Error occurred!", errorResult.data.error);
//            }
//        );
//    };
//}]);

File: server/application/src/main/java/pl/edu/agh/sogo/application/controller/UserController.java
package pl.edu.agh.sogo.application.controller;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;
import pl.edu.agh.sogo.domain.security.User;
import pl.edu.agh.sogo.service.IUserService;

import java.util.Collection;

@CrossOrigin(origins = "http://localhost:8080")
@RestController
@RequestMapping("/users")
public class UserController {
    @Autowired
    private IUserService userService;

    @ResponseBody
    @RequestMapping(method = RequestMethod.GET)
    public Collection<User> getUsers() {
        return userService.getUsers();
    }

    @ResponseBody
    @RequestMapping(value = "/{username}", method = RequestMethod.GET)
    public User getUser(@PathVariable(value = "username") String username) {
        return userService.findUserByUsername(username);
    }

    @ResponseBody
    @RequestMapping(method = RequestMethod.PUT)
    public void updateUser(@RequestBody User user) {
        userService.update(user);
        return;
    }

    @ResponseBody
    @RequestMapping(value = "/{username}/enable", method = RequestMethod.POST)
    public void enableUser(@PathVariable(value = "username") String username) {
        userService.enable(username);
    }

    @ResponseBody
    @RequestMapping(value = "/{username}/disable", method = RequestMethod.POST)
    public void disableUser(@PathVariable(value = "username") String username) {
        userService.disable(username);
    }

}

